### PR TITLE
Fix handling of nullable complex types

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Resolves https://github.com/Particular/NServiceBus/issues/5752

### CODE REVIEW

File: src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs [hunk 1]
```diff
@@ -46,16 +47,30 @@ public void CreateInstance_WhenMessageInitialized_ShouldBeThreadsafe()
...
         }
 
+#if NETCOREAPP
+#nullable enable
         [Test]
         public void Should_handle_messages_with_nullable_reference_types()
         {
             var mapper = new MessageMapper();
 
             // Type defined in separate assembly as a workaround
```
Comment (bording) [inline src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs:57]: This comment is no longer relevant.

File: src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs [hunk 2]
```diff
@@ -46,16 +47,30 @@ public void CreateInstance_WhenMessageInitialized_ShouldBeThreadsafe()
...
-            mapper.CreateInstance<WithDodgyNullable.IMyMessage>();
+            mapper.CreateInstance<IMessageWithNullableProperties>();
         }
 
+        public interface IMessageWithNullableProperties : ICommand, IMessage
+        {
+            string Schema { get; set; }
+
+            string? Notes { get; set; }
+            object[]? NullableComplexType { get; set; }
```
Comment (bording) [inline src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs:67]: This handles the first scenario mentioned in #5752, but what about https://github.com/Particular/NServiceBus/issues/5752#issuecomment-794065319?

Would be good to see that also in a test.

**Review by bording — CHANGES_REQUESTED** [review]

File: src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs (hunk 1 above)
Comment (SzymonPobiega) [reply src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs]: Fixed.

File: src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs (hunk 2 above)
Comment (SzymonPobiega) [reply src/NServiceBus.Core.Tests/MessageMapper/MessageMapperTests.cs]: Good catch. I added that, too.

**Review by ramonsmits — APPROVED** [review]

**Review by bording — APPROVED** [review]

**Pull request merged**